hazelcast/src/test/java/com/hazelcast/test/HzRunnerScheduler.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test;
+
+import org.junit.runners.model.RunnerScheduler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static com.hazelcast.test.HazelcastParallelClassRunner.getDefaultMaxThreads;
+
+/**
+ * Runner scheduler to run tests in parallel.
+ */
+public class HzRunnerScheduler implements RunnerScheduler {
```

Comment:
The JUnit manual mentions the `RunnerScheduler` abstraction as 
> still experimental, may go away.

Reply:
Yes, this is the weakest point of this PR. `RunnerScheduler` simplifies test parallelism but it may go away any time. What can be done? Maybe, I can introduce this runner in isolation from current runners and if `RunnerScheduler` goes away one day, we can also remove our implementation. Since `RunnerScheduler` exists since 2012, it may be ok to take that risk. Meanwhile we can gain some speed. WDYT?

Reply:
I agree, especially if it's been there for 8 years. Just wanted to raise awareness, I think it is safe to rely on that. Also, this can be fixed at a single place if they remove this abstraction (which I don't expect).